gef-ui/server/controllers/amendments/cover-end-date/validation.ts

```diff
@@ -0,0 +1,53 @@
+import { UnixTimestampSeconds, applyStandardValidationAndParseDateInput, DayMonthYearInput, COVER_END_DATE_MAXIMUM_YEARS_IN_FUTURE } from '@ukef/dtfs2-common';
+import { add, isAfter, isBefore, startOfDay, getUnixTime } from 'date-fns';
+import { ErrorsOrValue } from '../../../types/errors-or-value';
+import { mapValidationError } from '../../../utils/map-validation-error';
+
+/**
+ * @param {DayMonthYearInput} [params.dayMonthYear] - The cover end date as a DayMonthYearInput object
+ * @param {Date} [params.dayMonthYear] - The cover start date as Date
+ * @returns ErrorsOrValue<UnixTimestampSeconds> the value or errors depending on the validation result
+ */
```

Comment:
I dont think you want the types here as it overrides what TS thinks they are - we should only have one source of truth

```suggestion
/**
 * @param params.dayMonthYear - The cover end date as a DayMonthYearInput object
 * @param params.dayMonthYear - The cover start date as Date
 * @returns the value or errors depending on the validation result
 */
```